docs/editor/terminal-shell-integration.md

```diff
@@ -0,0 +1,173 @@
+- macOS: `/Applications/Visual\ Studio\ Code.app/Contents/Resources`
+
+**bash**
+
+  Add the following to your `~/.bashrc` file, replacing `<InstallDir>` with VS Code's installation directory. Run `code ~/.bashrc` in bash to open the file in VS Code.
+  ```sh
+  if [ "$TERM_PROGRAM" == "vscode" ]; then
+    . <InstallDir>/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
+  fi
+  ```
+**pwsh**
+
+  Add the following to your [PowerShell profile](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.2), replacing `<InstallDir>` with VS Code's installation directory. Run `code $Profile` in pwsh to open the file in VS Code.
+  ```pwsh
+  if ($env:TERM_PROGRAM -eq "vscode") {
+    . "<InstallDir>/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1"
+  }
+  ```
+**zsh**
+
+  Add the following to your `~/.zshrc` file. Run `code ~/.zshrc` in bash to open the file in VS Code.
+  ```sh
+  if [[ "$TERM_PROGRAM" == "vscode" ]]; then
+    . <vscode resources install path>/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh
+  fi
+  ```
+
+## Features
+
+### Command decorations and the overview ruler
+
+One of the things shell integration enables is the ability to get the exit codes of the commands run within the terminal. Using this information, decorations are added to the left of the line to indicate whether the command succeeded of failed. These decorations also show up in the relatively new overview ruler in the scroll bar, just like in the editor.
+
+![Blue circles appear next to successful commands, red circles with crosses appear next to failed commands. The color of the circles appears in the scroll bar](images/terminal-shell-integration/decorations.png)
+
+The decorations can be interacted with to give some contextual actions like re-running the command:
+
+![Clicking a successful command decoration shows a context menu containing items: Copy Output, Copy Output as HTML, Rerun Command and How does this work?]((images/terminal-shell-integration/decoration-menu.png)
+
+The command decorations can be configured with the following settings:
+
+- `terminal.integrated.shellIntegration.decorationIcon`
+- `terminal.integrated.shellIntegration.decorationIconSuccess`
+- `terminal.integrated.shellIntegration.decorationIconError`
+
+### Command navigation
+
+The commands detected by shell integration feed into the command navigation feature (`ctrl/cmd+up`, `ctrl/cmd+down`) to give it more reliable command positions. This feature enables quick navigation between commands and selection of their output.
+
+### Run recent command
+
+The `Terminal: Run Recent Command` command surfaces history from various sources in a quick pick, providing similar functionality to a shell's reverse search (ctrl+r). The sources are the current session's history, previous session history for this shell type and the common shell history file.
```

Comment:
should we mention that it has additional functionality like opening the output in an editor w a click?

Reply:
Good idea https://github.com/microsoft/vscode-docs/pull/5439/commits/7b9eebe7ab433ed5a06a838c5d85ff281e7bcb25

Also updated the release notes to include the update